core/ssh_keys_test.go

```diff
@@ -0,0 +1,476 @@
+				Fingerprint: sshFingerprint,
+			},
+			wantReq: &katapult.Request{
+				Method: "POST",
+				URL: &url.URL{
+					Path: "/core/v1/organizations/_/ssh_keys",
+					RawQuery: url.Values{
+						"organization[id]": []string{
+							"org_O648YDMEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+				Body: AuthSSHKeyProperties{
+					Name: "test",
+					Key:  sshPublicKey,
+				},
+			},
+			wantResp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ctx: context.Background(),
+				org: OrganizationRef{ID: "org_O648YDMEYeLmqdmn"},
+				args: AuthSSHKeyProperties{
+					Name: "test",
+					Key:  sshPublicKey,
+				},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			tc := testclient.New(tt.resp, tt.respErr, tt.respV)
+			c := NewSSHKeysClient(tc)
+			ctx := test.Context(tt.args.ctx)
+
+			got, resp, err := c.Add(ctx, tt.args.org, tt.args.args)
+
+			assert.Equal(t, 1, len(tc.Calls), "only 1 request should be made")
+			test.AssertContext(t, ctx, tc.Ctx)
+
+			assert.Equal(t, tt.want, got)
+
+			if tt.wantResp != nil {
+				assert.Equal(t, tt.wantResp, resp)
+			}
+
+			if tt.wantReq != nil {
+				assert.Equal(t, tt.wantReq, tc.Request)
+			}
+
+			if tt.wantErr == "" {
+				assert.NoError(t, err)
+			} else {
+				assert.EqualError(t, err, tt.wantErr)
+			}
+		})
+	}
+}
+
+func TestSSHKeysClient_Delete(t *testing.T) {
+	type args struct {
+		ctx context.Context
+		ref SSHKeyRef
+	}
+	tests := []struct {
+		name     string
+		args     args
+		resp     *katapult.Response
+		respErr  error
+		respV    *sshKeysResponseBody
+		want     *AuthSSHKey
+		wantReq  *katapult.Request
+		wantResp *katapult.Response
+		wantErr  string
+	}{
+		{
+			name: "success",
+			args: args{
+				ctx: context.Background(),
+				ref: SSHKeyRef{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			respV: &sshKeysResponseBody{
+				SSHKey: &AuthSSHKey{
+					ID:          "testing-id",
+					Name:        "test",
+					Fingerprint: sshFingerprint,
+				},
+			},
+			resp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+			want: &AuthSSHKey{
+				ID:          "testing-id",
+				Name:        "test",
+				Fingerprint: sshFingerprint,
+			},
+			wantReq: &katapult.Request{
+				Method: "DELETE",
+				URL: &url.URL{
+					Path: "/core/v1/ssh_keys/_",
+					RawQuery: url.Values{
+						"ssh_key[id]": []string{
+							"ssh_O574YEEEYeLmqdmn",
+						},
+					}.Encode(),
+				},
+			},
+			wantResp: &katapult.Response{
+				Response: &http.Response{StatusCode: http.StatusOK},
+			},
+		},
+		{
+			name: "request error",
+			args: args{
+				ctx: context.Background(),
+				ref: SSHKeyRef{ID: "ssh_O574YEEEYeLmqdmn"},
+			},
+			respErr: fmt.Errorf("flux capacitor undercharged"),
+			wantErr: "flux capacitor undercharged",
+		},
```

Comment:
Annoying as it may be to duplicate, I think each of the functions should have failure tests for both with and without a response object being returned.